lib/dealership.rb

```diff
@@ -0,0 +1,57 @@
+require './lib/car'
+
+class Dealership
+
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+  end
+
+  def inventory_count
+    @inventory.length
+  end
+
+  def add_car(car)
+    @inventory.push(car)
+    return nil
+  end
+
+  def has_inventory?
+    @inventory.length >= 1
+  end
+
+  def cars_by_make(make)
+    @inventory.find_all do |car|
+      make == car.make
+    end
+  end
+
+  def total_value
+    total_value = 0
+    @inventory.each do |car|
+      total_value += car.total_cost
+    end
+    total_value
+  end
+
+  def details
+    {
+      "total_value" => self.total_value,
+      "address" => self.address
+    }
+  end
+
+  def average_price_of_car
+    (self.total_value / @inventory.length).to_s
```

Comment:
Was hoping to have the comma in the returned value here. It's definitely a challenge! Here is some further reading of one option of how to do it https://rick-moore.medium.com/formatting-number-strings-in-ruby-4da35d5282e3